site/app/views/grading/ElectronicGraderView.php

```diff
@@ -1595,8 +1595,28 @@ function display_files($files, &$count, $indent, &$return, $filename) {
 </div>
 
 <script type="text/javascript">
+var __unloadSentRequest = false;
 window.onbeforeunload = function() {
-    saveLastOpenedMark('{$gradeable->getId()}' ,'{$user->getAnonId()}', {$gradeable->getActiveVersion()}, '{$your_user_id}', '-1', false);
+    // Will only work for Chrome
+    if (!__unloadSentRequest) {
+        console.log("before unload");
+        saveLastOpenedMark('{$gradeable->getId()}' ,'{$user->getAnonId()}', {$gradeable->getActiveVersion()}, '{$your_user_id}', '-1', false, function() {
+            __unloadSentRequest = true;
+        }, function() {
+            // don't error
+        });
+    }
+}
+window.onunload = function() {
+    // Will work for other browsers
+    if (!__unloadSentRequest) {
+        console.log("unload");
+        saveLastOpenedMark('{$gradeable->getId()}' ,'{$user->getAnonId()}', {$gradeable->getActiveVersion()}, '{$your_user_id}', '-1', false, function() {
+            __unloadSentRequest = true;
+        }, function() {
+            // don't error
+        });
+    }
 }
```

Comment:
Can you make this a function and you should set `__unloadSentRequest` (though rename this to `__unloadRequestSent`) immediately once entering the if statement just to better avoid any potential race conditions.

```
function unloadPage() {
    if (!__unloadRequestSent) {
        __unloadRequestSent = true;
        ...
    }
}
```

Call the function from `window.onunload` and `window.onbeforeunload`. Please also add a docstring above this stuff to explain why we're doing it with any potential links to resources that explain what's going on just for the future developer who might wonder what's going on here.